features/ifttt/src/main/java/org/opennms/features/ifttt/IfTttDaemon.java

```diff
@@ -0,0 +1,349 @@
+
+                                final OnmsSeverity newSeverity = maxAlarmsSeverity.orElse(OnmsSeverity.NORMAL);
+                                final int newAlarmCount = filteredAlarms.size();
+
+                                LOG.debug("Received {} filtered, {} new severity", newAlarmCount, newSeverity);
+
+                                final DefaultVariableNameExpansion defaultVariableNameExpansion = new DefaultVariableNameExpansion(
+                                        oldSeverity.get(triggerPackage.getOnlyUnacknowledged()).get(triggerPackage.getCategoryFilter()), newSeverity,
+                                        oldAlarmCount.get(triggerPackage.getOnlyUnacknowledged()).get(triggerPackage.getCategoryFilter()), newAlarmCount
+                                );
+
+                                // Trigger IFTTT event if necessary.
+
+                                if (!newSeverity.equals(oldSeverity.get(triggerPackage.getOnlyUnacknowledged()).get(triggerPackage.getCategoryFilter())) ||
+                                        newAlarmCount != oldAlarmCount.get(triggerPackage.getOnlyUnacknowledged()).get(triggerPackage.getCategoryFilter())) {
+                                    fireIfTttTriggerSet(ifTttConfig, triggerPackage.getCategoryFilter(), newSeverity, defaultVariableNameExpansion);
+                                }
+
+                                LOG.debug("Old severity: {}, new severity: {}, old alarm count: {}, new alarm count: {}",
+                                        oldSeverity.get(triggerPackage.getOnlyUnacknowledged()).get(triggerPackage.getCategoryFilter()), newSeverity,
+                                        oldAlarmCount.get(triggerPackage.getOnlyUnacknowledged()).get(triggerPackage.getCategoryFilter()), newAlarmCount
+                                );
+
+                                oldSeverity.get(triggerPackage.getOnlyUnacknowledged()).put(triggerPackage.getCategoryFilter(), newSeverity);
+                                oldAlarmCount.get(triggerPackage.getOnlyUnacknowledged()).put(triggerPackage.getCategoryFilter(), newAlarmCount);
+                            }
+                        }
+                    });
+                } catch (Exception e) {
+                    LOG.error("Error while polling alarm table.", e);
+                } finally {
+                    LOG.debug("Run complete. Next poll in {}s.", pollInterval);
+                }
+            }
+        }, pollInterval, pollInterval, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Restarts the poller with the given polling interval. This method is triggered by changes of the pollInterval in
+     * the configuration file.
+     *
+     * @param pollInterval the polling interval to be used
+     */
+    private void restartPoller(final long pollInterval) {
+        LOG.debug("Restarting alarm poller (interval {}s).", pollInterval);
+
+        if (m_alarmPoller != null) {
```

Comment:
Stopping the poller also does check for the poller to be not null. I'd be consistent about this. 
Either check in stopPoller or before.